### ISSUE TITLE
HA 2023.7 with Tuya-Local 2023.6.1 has removed Orion BLE Lock Auto-Lock entity

### ISSUE THREAD

**Issue by jaydisc** [make-all/tuya-local#870]
Since updating to HA 2023.7, the entity switch for Auto-Lock is reported as unavailable:

> "This entity is no longer being provided by the tuya_local integration. If the entity is no longer in use, delete it in settings."

Updating TL to 2023.6.1 made no noticeable difference. I had previously been running https://github.com/make-all/tuya-local/commit/b9d978a8a623c6cd9a0fb4864df5607718451826.

I have attached a debug log fresh off of a reboot, as well as device diagnostics for two of these locks:

[Debug Log](https://github.com/make-all/tuya-local/files/11963457/tuya_local.log)
[tuya_local-d9af722e59b6b9a1f660f2c84e662b96-Office Lock Living Room](https://github.com/make-all/tuya-local/files/11963455/tuya_local-d9af722e59b6b9a1f660f2c84e662b96-Office.Lock.Living.Room-072552c49aee8b6da5b899b4afc6f1f1.json.txt)
[tuya_local-baf7d32887458f9fd5db882b9b772986-Office Lock Hallway](https://github.com/make-all/tuya-local/files/11963456/tuya_local-baf7d32887458f9fd5db882b9b772986-Office.Lock.Hallway-3edd6a86e47973b87580caf1d3284f95.json-2.txt)

**Comment by jaydisc** [comment]
I had been blocking the Tuye Bluetooth Hub from access the internet at the router level (a big benefit of Tuya-local?). I noticed that in the Tuya app, which was working, it was only working via Bluetooth (reported locks offline when I disabled Bluetooth), and as expected, the Bluetooth hub was reported as offline.

So, I unblocked the Bluetooth Hub at the router and rebooted the hub, and things are now working, albeit a bit differently:

1. The Auto-Lock entity is indeed still gone, BUT… the original Lock entity (which had previously not been functional) is now working just as the Auto-Lock previously did! 👍
2. AND… an automation trigger based on that lock entity is now working! (I will separately update https://github.com/make-all/tuya-local/issues/798)

I didn't expect to have issue blocking the Bluetooth Hub's internet. As best as I can guess, it appears that when this happened, the hub "released" (?) the two locks which then reverted to direct Bluetooth communication? Now that I have upgraded, and it's all working again, I will try to re-disable the Hub's internet.
